src/internal/rest.rs

```diff
@@ -36,69 +44,325 @@ pub struct ServerError {
     code: u32,
 }
 
+define_encode_set! {
+    /// This encode set should be used for path components and query strings.
+    /// `A-Z a-z 0-9 - _ . ! ~ * ' ( )` are the only characters we _don't_ want to encode.
+    ///
+    /// If this is changed it will potentially need be changed in the webservice and all other SDKs.
+    #[rustfmt::skip]
+    pub ICL_ENCODE_SET = [SIMPLE_ENCODE_SET] | {
```

Comment:
I don't disagree. I don't know how to specify a blacklist EncodeSet that works with the `percentencoding`. Maybe there's a way?

Reply:
I think maybe removing all the commented out `do not encode` lines would help make it more digestable.

Reply:
Since the comment refers to the items marked as "do not encode" I think they are important. The only way I could make sure to cover everything was to print the ASCII table and go through it character by character. I did remove a bunch of the letters and numbers to make the list much shorter. Hopefully that's a decent compromise.